### ISSUE TITLE
Style/FrozenStringLiteralComment matches arbitrary token as being no offense

### ISSUE THREAD

**Issue by jonathanschlue-as** [rubocop-hq/rubocop#7241]
Rubocop's cop `Style/FrozenStringLiteralComment` for enforcing the so called frozen string literal comment misses some offenses.

It does not report offenses if an arbitrary string value without whitespace is specified, although the resulting magic comment does not switch on frozen string literal behaviour.

Consider the file `foobar.rb`:

```rb
# frozen_string_literal: foobar

puts "some string".frozen?
```

```shell
$ ruby foobar.rb
false
```

--------

## Expected behavior

It should recognize magic comments defining values for `frozen_string_literal` that do not actually switch on frozen string literal behaviour as an offense.

## Actual behavior

It does not recognize the magic comment

```
# frozen_string_literal: foobar
```

as an offense.

## Steps to reproduce the problem

1. Create file `foobar.rb` containing:

```
# frozen_string_literal: foobar

puts "some string".frozen?
```

2. Enable the `Style/FrozenStringLiteralComment` cop.
3. Run `rubocop foobar.rb`.

## RuboCop version

```
$ bundle exec rubocop -V
0.73.0 (using Parser 2.6.3.0, running on ruby 2.5.5 x86_64-linux)
```
